main.rb

```diff
@@ -0,0 +1,130 @@
+# rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(__method__) unless block_given?
+
+    arr_len = Array(self).length
+    arr_len.times do |index|
+      yield(Array(self)[index])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(__method__) unless block_given?
+
+    arr_len = Array(self).length
+    arr_len.times do |index|
+      yield(Array(self)[index], index)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(__method__) unless block_given?
+
+    new_arr = []
+    my_each { |item| new_arr << item if yield(item) }
+    new_arr
+  end
+
+  def my_all?(arg = nil)
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |block| return false unless block.is_a? arg }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |block| return false unless block == arg }
+    elsif !arg.nil? && arg.is_a?(String) || arg.is_a?(Regexp)
+      my_each { |block| return false unless block.match(arg) }
+    elsif !block_given?
+      my_each { |block| return false if block.nil? || !block }
+    else
+      my_each { |block| return false unless yield(block) }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |block| return true if block.is_a? arg }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |block| return true if block == arg }
+    elsif !arg.nil? && arg.is_a?(String) || arg.is_a?(Regexp)
+      my_each { |block| return true if block.match(arg) }
+    elsif !block_given?
+      my_each { |block| return true unless block.nil? || !block }
+    else
+      my_each { |block| return true if yield(block) }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |block| return false if block.is_a? arg }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |block| return false if block == arg }
+    elsif !arg.nil? && arg.is_a?(String) || arg.is_a?(Regexp)
+      my_each { |block| return false if block.match(arg) }
+    elsif !block_given?
+      my_each { |block| return false unless block.nil? || !block }
+    else
+      my_each { |block| return false if yield(block) }
+    end
+    true
+  end
```

Comment:
#### [OPTIONAL]
I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
def my_none?(arg = nil, &block)
  !my_any?(arg, &block)
end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?